L80-csharp-load-balancing.md

```diff
@@ -0,0 +1,259 @@
+        public void Set<TValue>(BalancerAttributesKey<TValue> key, TValue value);
+        public bool TryGetValue<TValue>(BalancerAttributesKey<TValue> key, out TValue value);
+        void ICollection<KeyValuePair<string, object>>.Add(KeyValuePair<string, object?> item);
+        void ICollection<KeyValuePair<string, object>>.Clear();
+        bool ICollection<KeyValuePair<string, object>>.Contains(KeyValuePair<string, object?> item);
+        void ICollection<KeyValuePair<string, object>>.CopyTo(KeyValuePair<string, object?>[] array, int arrayIndex);
+        bool ICollection<KeyValuePair<string, object>>.Remove(KeyValuePair<string, object?> item);
+        void IDictionary<string, object>.Add(string key, object? value);
+        bool IDictionary<string, object>.ContainsKey(string key);
+        object? IDictionary<string, object>.get_Item(string key);
+        bool IDictionary<string, object>.Remove(string key);
+        void IDictionary<string, object>.set_Item(string key, object? value);
+        bool IDictionary<string, object>.TryGetValue(string key, out object? value);
+        IEnumerator<KeyValuePair<string, object?>> IEnumerable<KeyValuePair<string, object>>.GetEnumerator();
+        IEnumerator IEnumerable.GetEnumerator();
+    }
+    public readonly struct BalancerAttributesKey<TValue> {
+        public BalancerAttributesKey(string key);
+        public string Key { get; }
+    }
+    public class BalancerState {
+        public BalancerState(ConnectivityState connectivityState, SubChannelPicker picker);
+        public ConnectivityState ConnectivityState { get; }
+        public SubChannelPicker Picker { get; }
+    }
+    public class ChannelState {
+        public ChannelState(AddressResolverResult resolverState, BalancerAttributes options);
+        public BalancerAttributes Options { get; set; }
+        public AddressResolverResult ResolverState { get; set; }
+    }
+    public class CompleteContext {
+        public CompleteContext();
+        public Exception? Error { get; set; }
+        public Metadata? Trailers { get; set; }
+    }
+    public class DnsAddressResolver : AddressResolver {
+        public DnsAddressResolver(Uri address, ILoggerFactory loggerFactory);
+        public override Task RefreshAsync(CancellationToken cancellationToken);
+        public override IDisposable Subscribe(IObserver<AddressResolverResult> observer);
+        protected override void Dispose(bool disposing);
+    }
+    public class DnsAddressResolverFactory : AddressResolverFactory {
+        public DnsAddressResolverFactory(ILoggerFactory loggerFactory);
+        public override string Name { get; }
+        public override AddressResolver Create(Uri address, AddressResolverOptions options);
+    }
+    public abstract class LoadBalancer : IDisposable {
+        protected LoadBalancer();
+        public void Dispose();
+        public abstract void ResolverError(Exception exception);
```

Comment:
Removed `ResolverError` and merged its logic into `UpdateChannelState`.

However, I'd prefer to leave `UpdateChannelState` returning void at the moment. Since I'm not a load balancing expert I'd like to be conservative and follow the existing pattern instead of blazing new ground. Once other implementations do this then `UpdateChannelState` can be updated with this new logic.